impl/src/main/java/org/ehcache/internal/store/DefaultStoreProvider.java

```diff
@@ -60,6 +60,11 @@
       if (heapPool == null) {
         throw new IllegalArgumentException("Cannot store to disk without heap resource");
       }
+      if (diskPool.getSize() == heapPool.getSize()) {
+        LOG.info("Disk pool size should be greater than Heap pool size");
+      } else if (diskPool.getSize() < heapPool.getSize()) {
+        throw new IllegalArgumentException("Disk pool size must be greater than Heap pool size");
+      }
```

Comment:
I'm fairly confident this could already be implemented agnostically from the existing limitation imposed on the tiering model, i.e. in a way that the pyramid gets validated so that higher tiers are always smaller than lower ones that share the common `ResourceUnit`... 
Doing it that way, would make sure this constraint remains enforced when we do ARC or introduce the new `DiskStore`.